comment_limit.module

```diff
@@ -62,42 +62,57 @@ function comment_limit_form_field_config_edit_form_alter(&$form, FormStateInterf
 }
 
 /**
- * Implements hook_form_alter().
+ * Implements hook_form_FORM_ID_alter().
  */
-function comment_limit_form_alter(&$form, FormStateInterface $form_state) {
-  if ($form['#id'] == 'comment-form') {
-    $comment = $form_state->getFormObject()->getEntity();
-    $comment_limit = Drupal::service('comment_limit.service');
-    $form_action = explode('/', $form['#action']);
-    $last_key = count($form_action) - 1;
-    $field_name = $form_action[$last_key];
-    $entity_id = $comment->getCommentedEntityId();
-    $entity_type = $comment->getCommentedEntityTypeId();
-    $entity = Drupal::entityTypeManager()
-      ->getStorage($entity_type)
-      ->load($entity_id);
-    $entity_bundle = $entity->bundle();
-    $field_config = FieldConfig::loadByName($entity_type, $entity_bundle, $field_name);
-    $field_id = $field_config->id();
-    $field_label = $field_config->getLabel();
-    $definitions = DataDefinition::create('any')
-      ->addConstraint('CommentFormConstraint', [
-        'entityId' => $entity_id,
-        'entityType' => $entity_type,
-        'fieldId' => $field_id,
-        'fieldName' => $field_name,
-      ]);
-    $typed_data = Drupal::typedDataManager()->create($definitions);
-    $violations = $typed_data->validate();
-
-    if ($violations->count() > 0) {
-      $form = [];
-      $form[$field_name] = [
-        '#type' => 'html_tag',
-        '#tag' => 'b',
-        '#value' => $comment_limit->getMessage($entity_id, $entity_type, $field_name, $field_id, $field_label),
-      ];
+function comment_limit_form_comment_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $comment_limit = Drupal::service('comment_limit.service');
+  $comment = $form_state->getFormObject()->getEntity();
+
+  $form_action_parts = explode('/', $form['#action']);
+  $field_name = $form_action_parts[count($form_action_parts) - 1];
+
+  $entity = Drupal::entityTypeManager()
+    ->getStorage($comment->getCommentedEntityTypeId())
+    ->load($comment->getCommentedEntityId());
+
+  $field_config = FieldConfig::loadByName(
+    $comment->getCommentedEntityTypeId(),
+    $entity->bundle(),
+    $field_name
+  );
+  $field_id = $field_config->id();
+  $field_label = $field_config->getLabel();
+
+  $definitions = DataDefinition::create('any')
+    ->addConstraint('CommentFormConstraint', [
+      'entityId' => $comment->getCommentedEntityId(),
+      'entityType' => $comment->getCommentedEntityTypeId(),
+      'fieldId' => $field_id,
+      'fieldName' => $field_name,
+    ]);
+  $typed_data = Drupal::typedDataManager()->create($definitions);
+  $violations = $typed_data->validate();
+  if ($violations->count() > 0) {
+    // Check access of the form elements.
+    $access = \Drupal::currentUser()->hasPermission('bypass comment limit');
```

Comment:
If you like to check here on user permission than you should change the validation in getMessage() also.